docs/markdown/Writing Plugins/tutorials/01-create-new-goal-with-a-custom-target.md

```diff
@@ -0,0 +1,364 @@
+
+## Registering a plugin
+
+We'll be writing an [in-repo plugin](https://www.pantsbuild.org/docs/plugins-overview#in-repo-plugins), and expect 
+you to have the `pants-plugins/project_version` directory as well as the `pants.toml` file with this configuration:
+
+```toml pants.toml
+# Specifying the path to our plugin's top-level folder using the `pythonpath` option:
+pythonpath = ["%(buildroot)s/pants-plugins"]
+
+backend_packages = [
+  "pants.backend.python",
+   ...
+  "project_version",
+]
+```
+
+## Creating a new target
+
+Once you have become familiar with the [core concepts of Targets and Fields](https://www.pantsbuild.org/docs/target-api-concepts), you are ready to [create an own custom target](https://www.pantsbuild.org/docs/target-api-new-targets) that will represent the `VERSION` file:
+
+```python pants-plugins/project_version/targets.py
+from pants.engine.target import COMMON_TARGET_FIELDS, SingleSourceField, Target
+
+
+class ProjectVersionTarget(Target):
+    alias = "version_file"
+    core_fields = (*COMMON_TARGET_FIELDS, SingleSourceField)
+    help = "A project version target representing the VERSION file."
+```
+
+Our target has some common target fields such as `tags` and `description` available via the `COMMON_TARGET_FIELDS`; including those fields in your targets may be convenient if you decide to use tags and provide a description later. In addition, it also has the [`source` field](https://www.pantsbuild.org/docs/rules-api-and-target-api#sourcesfield) which will be used to provide path to the project's `VERSION` file. 
+
+We could [add a custom field](https://www.pantsbuild.org/docs/target-api-new-fields) to provide a file path, however, there are multiple advantages to  using the `source` field. You will learn more about them in the following tutorials. 
+
+In order to start using a target, you only need to register it: 
+
+```python pants-plugins/project_version/register.py
+from typing import Iterable
+
+from pants.engine.target import Target
+from project_version.target_types import ProjectVersionTarget
+
+
+def target_types() -> Iterable[type[Target]]:
+    return [ProjectVersionTarget]
+```
+
+You can now run `./pants help version_file` to learn more about the target. You can now also add a target to a `myapp/BUILD` file:
```

Comment:
So, ripping off my own post about writing plugins (https://sureshjoshi.com/development/first-pants-plugin#our-first-plugin) - whenever I state a command, I always (try to) show some console output. 

That way, if a user is typing along, they can make sure that each step is working as expected, and do micro modifications, and catch problems as early as possible.

e.g. 
```bash
./pants help fooify

\`fooify\` target
---------------

The \`fooify\` target will take in a wheel dependency and add a .foo extension to the end.

Valid fields:

[...]
```

Reply:
Great idea, done!